inference: enable `:call` inference in irinterp

aviatesk opened the pull request:
Built on top of #48913, this commit enables `:call` inference in irinterp. In a case when some regression is detected, we can simply revert this commit rather than reverting the whole refactoring from #48913.

@nanosoldier `runbenchmarks("inference", vs=":master")`
@nanosoldier `runtests()`

aviatesk committed:
inference: enable `:call` inference in irinterp

Built on top of #48913, this commit enables `:call` inference in irinterp.
In a case when some regression is detected, we can simply revert this
commit rather than reverting the whole refactoring from #48913.

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/49191) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/d626e84_vs_da1ce65/report.md).

aviatesk commented:
@nanosoldier `runtests()`

aviatesk commented:
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`
@nanosoldier `runtests()`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1501777431) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/fa46f63_vs_94da492/report.md).

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1501777431) has completed - possible new issues were detected.
A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/ab3fee7_vs_7ddb4e5/report.html).

aviatesk commented:
@nanosoldier `runbenchmarks("inference", vs=":master")`
@nanosoldier `runtests()`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1526944278) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/3e22e40_vs_f11bfc6/report.md).

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1526944278) has completed - possible new issues were detected.
A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/e891443_vs_f11bfc6/report.html).

aviatesk commented:
@nanosoldier `runtests(["InformationGeometry", "KissMCMC", "Braket", "MPIMeasurements", "OutlierDetectionPython", "ClimaCore", "DifferentialForms", "LiteQTL", "Trixi2Vtk", "SubSIt", "Controlz", "Relief", "TensorRefinement"])`

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1527433164) has completed - possible new issues were detected.
A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/e891443_vs_f11bfc6/report.html).

aviatesk committed:
fix irinterp lattice

Now `LimitedAccuracy` can appear in irinterp, so we should include
`InferenceLattice` for `[typeinf|ipo]_lattice` for irinterp.

aviatesk commented:
The failure of `TensorRefinement` was real. Now `LimitedAccuracy` can appear in irinterp, so we should include `InferenceLattice` for `[typeinf|ipo]_lattice` for irinterp. [0e822cd](https://github.com/JuliaLang/julia/pull/49191/commits/0e822cda800a2ec2e0cf40971cd92824eb68e1a8) fixes it.

aviatesk commented:
@nanosoldier `runbenchmarks("inference", vs=":master")`
@nanosoldier `runtests()`

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1527688190) has completed - possible new issues were detected.
A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/0e822cd_vs_cc7fec7/report.html).

aviatesk commented:
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`
@nanosoldier `runtests(["EBayes", "DelaunayTriangulation", "Permanents", "OteraEngine", "ScikitLearn", "ASE", "LazySets", "SBMLToolkit", "Causal", "Decapodes", "Relief", "SubSIt"])`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1528821492) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/81e2163_vs_219dc10/report.md).

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1528821492) has completed - possible new issues were detected.
A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/a8d99f7_vs_219dc10/report.html).

aviatesk commented:
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`
@nanosoldier `runtests(["EBayes", "DelaunayTriangulation", "Permanents", "OteraEngine", "ScikitLearn", "ASE", "LazySets", "SBMLToolkit", "Causal", "Decapodes", "Relief", "SubSIt"])`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1534216899) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/1050647_vs_827d34a/report.md).

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/49191#issuecomment-1534216899) has completed - no new issues were detected.
A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/9436bf5_vs_827d34a/report.html).
